services/movie.py

```diff
@@ -0,0 +1,51 @@
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: list[int] | None = None,
+        actors_ids: list[int] | None = None
+) -> Movie | QuerySet:
+
+    query = Movie.objects.all()
+
+    if actors_ids:
+        query = query.filter(
+            actors__id__in=actors_ids,
+        )
+
+    if genres_ids:
+        query = query.filter(
+            genres__id__in=genres_ids
+        )
+
+    return query.distinct()
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    try:
+        movie = Movie.objects.get(id=movie_id)
+        return movie
+    except Movie.DoesNotExist:
+        print("Enter valid Movie id")
```

Comment:
If you want to catch the case if Movie does not exist by some id . Use `get_object_or_404` function which comes from Django by default 
`from django.shortcuts import get_object_or_404
`